`AutoComplete::ForUser` — restrict to verified users

nimmolo opened the pull request:
Also adds User class method `unique_text_name` for combined name attributes, because the instance method is no longer available in AutoComplete when we use AR to select the columns we need (they're not instances of User anymore).

nimmolo committed:
Add user `verified` and `unverified` scopes

nimmolo committed:
Fix user auto_complete to check verification, change format to `unique_format_name`

nimmolo committed:
Keep login first so the returned ordering makes sense

nimmolo committed:
Modifications to use class method

nimmolo committed:
Switch to unique_text_name

nimmolo committed:
Update for_user.rb

nimmolo committed:
Update autocompleters_controller_test.rb

nimmolo committed:
Update autocompleters_controller_test.rb

nimmolo committed:
Update field_slips_controller_test.rb

mo-nathan commented on app/classes/auto_complete/for_user.rb:
```diff
@@ -28,10 +28,10 @@ def matches_array(users)
                     else
-                      "#{user[:login]} <#{user[:name]}>"
+                      User.unique_text_name(user)
                     end
```
Why are you doing this rather than `user.unique_text_name`?

mo-nathan replied:
Proposed changes here: https://github.com/MushroomObserver/mushroom-observer/pull/2665

nimmolo replied:
Thank you

mo-nathan committed:
Update for_user.rb

I think this is equivalent and allows us to get rid of the new redundant class function in user.rb.

mo-nathan committed:
Fix weird whitespace issue

nimmolo committed:
Merge pull request #2665 from MushroomObserver/mo-nathan-patch-1

Update handling of unique_text_name
